src/DotNetStatus/JsonData/JsonData.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Fx.Portability
+{
+    public class JsonData
+    {
+        private HttpClient _client;
+        private readonly string _apiPortUrl;
+
+        public JsonData(string apiPortUrl, ProductInformation productInfo)
+        {
+            _client = CreateHttpClient();
+            _apiPortUrl = apiPortUrl;
+        }
+
+        /// <summary>
+        /// Returns Json from the WebAPI calls made to the APIPortService
+        /// </summary>
+        /// <param name="query">the webapi action to run</param>
+        public async Task<T> GetJsonDataAsync<T>(string query)
+        {
+            using (var response = await _client.GetAsync(CreateFullUrl(query)))
+            {
+                if (!response.IsSuccessStatusCode)
+                {
+                    return default(T);
+                }
+
+                var content = await response.Content.ReadAsStringAsync();
+                var result = JsonConvert.DeserializeObject<T>(content, DataExtensions.JsonSettings);
+                return result;
+            }
+        }
+
+        /// <summary>
+        /// Creates an HTTPClient
+        /// </summary>
+        /// <returns></returns>
+        private HttpClient CreateHttpClient()
+        {
+            var client = new HttpClient();
```

Comment:
Look at the BaseUrl property of HttpClient - that would allow you not to need to track _apiPortServiceUrl